sql-plugin/src/main/scala/com/nvidia/spark/rapids/datetimeRebaseUtils.scala

```diff
@@ -23,9 +23,88 @@ import com.nvidia.spark.rapids.shims.SparkShimImpl
 import org.apache.spark.sql.catalyst.util.RebaseDateTime
 import org.apache.spark.sql.rapids.execution.TrampolineUtil
 
-object RebaseHelper {
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy
+ */
+sealed abstract class DateTimeRebaseMode(val value: String) extends Serializable
+
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy.EXCEPTION
+ */
+case object DateTimeRebaseException extends DateTimeRebaseMode("EXCEPTION")
+
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy.LEGACY
+ */
+case object DateTimeRebaseLegacy extends DateTimeRebaseMode("LEGACY")
+
+/**
+ * Mirror of Spark's LegacyBehaviorPolicy.CORRECTED
+ */
+case object DateTimeRebaseCorrected extends DateTimeRebaseMode("CORRECTED")
+
+object DateTimeRebaseUtils {
+  def invalidRebaseModeMessage(name: String): String =
+    s"Invalid datetime rebase mode: $name (must be either 'EXCEPTION', 'LEGACY', or 'CORRECTED')"
+
+  def rebaseModeFromName(name: String): DateTimeRebaseMode = name match {
```

Comment:
Nit: I think this would be cleaner as a `fromName` method provided by a DateTimeRebaseMode object.